### PR TITLE
apply cmake mods for hunter

### PR DESCRIPTION
**Pull request by headupinclouds**
* huntergate etc
* pkgconfig installation
* option for BUILD_TESTS and BUILD_UNIT_TESTS
* add version 1.0.0

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
+cmake_minimum_required(VERSION 3.0)
 
-project(dmlc C CXX)
+include("cmake/HunterGate.cmake")
+HunterGate(
+  URL "https://github.com/ruslo/hunter/archive/v0.19.210.tar.gz"
+  SHA1 "87eb902de6ca89e1dafc71e475344f523ef67729"
+)
+
+project(dmlc VERSION 1.0.0 LANGUAGES  C CXX)
```
Comment (ruslo) [inline CMakeLists.txt]: `LANGUAGES C CXX` is a default, can be skipped.

File: cmake/Config.cmake.in [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+endif()
+
+if(@USE_OPENMP@)
+  if(NOT @MSVC@)
+    find_package(OpenMP REQUIRED)
+  endif()
+endif()
+
+if(@MSVC@)
+  find_package(PythonInterp)
```
Comment (ruslo) [inline cmake/Config.cmake.in]: This one not needed since it's not a dependency requirement.

File: .gitignore [hunk 3]
```diff
@@ -45,3 +45,7 @@ build
 
 # CLion
 .idea
+
+_logs
```
Comment (ruslo) [inline .gitignore]: I think it's possible to use glob `_*`

File: CMakeLists.txt [hunk 4]
```diff
@@ -1,6 +1,12 @@
-cmake_minimum_required(VERSION 2.8.7)
+cmake_minimum_required(VERSION 3.0)
```
Comment (ruslo) [inline CMakeLists.txt]: `Threads::Threads` available since CMake 3.1

Comment (headupinclouds) [reply CMakeLists.txt]: done

File: CMakeLists.txt (hunk 1 above)
Comment (headupinclouds) [reply CMakeLists.txt]: done

File: cmake/Config.cmake.in (hunk 2 above)
Comment (headupinclouds) [reply cmake/Config.cmake.in]: done

File: .gitignore (hunk 3 above)
Comment (headupinclouds) [reply .gitignore]: done

**Comment by headupinclouds** [comment]
Changes applied.  Thanks for the review!

**Pull request merged**